src/main/java/com/salesforce/storm/spout/dynamic/kafka/Consumer.java

```diff
@@ -751,4 +771,16 @@ public void removeConsumerState() {
         // Return TopicPartitions for our assigned partitions
         return topicPartitions;
     }
+
+    /**
+     * Utility method to reduce duplication around generating metric keyspaces for
+     * topic/partition metrics.
+     * @param topic Name of topic the metric belongs to.
+     * @param partition Partition the metric belongs to.
+     * @param metric Name of the metric.
+     * @return Properly formatted keyspace for metric.
+     */
+    private String getTopicPartitionMetricKey(final String topic, final int partition, final String metric) {
+        return "topic." + topic + ".partition." + partition + "." + metric;
```

Comment:
I think we should use String.format() here.

Reply:
For readability I 100% agree.  For performance, some guy somewhere out on the internet claims its significantly slower compared to just straight concatenation.  

Unsure the best approach here.  Going to merge this in as is to your branch, and will toy with it in your branch.